cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -323,10 +323,14 @@ impl YaccParser {
         let mut errs: Vec<YaccGrammarError> = Vec::new();
         let mut result = self.parse_grmtools_header(0);
 
-        if let YaccKind::SelfDescribing = self.yacc_kind {
+        if let YaccKind::SelfDescribing(default) = &self.yacc_kind {
```

Comment:
One minor thing occurred to me,

This default/fallback setting code should probably be done in a loop, 
otherwise silly values can cause this to reach later panics with
`SelfDescribing(SelfDescribing(None))`, strangely making it a while causes an infinite loop. 
I don't yet understand why since I thought it should be decreasing/well founded.

Reply:
Fixed in 19b6629